lib/intercom/request.rb

```diff
@@ -152,6 +152,8 @@ def raise_application_errors_on_failure(error_list_details, http_code)
       case error_code
       when 'unauthorized', 'forbidden', 'token_not_found'
         raise Intercom::AuthenticationError.new(error_details['message'], error_context)
+      when 'token_suspended', 'token_revoked', 'token_unauthorized'
```

Comment:
So my thought here was to introduce two new errors for `token_revoked` and `token_unauthorized`. Or we can introduce one new error which is generic that can be `TokenError`. Your call 👍

Reply:
🤦‍♂️